lldb/cmake/caches/Apple-lldb-Linux.cmake

```diff
@@ -7,3 +7,7 @@ set(LLVM_DISTRIBUTION_COMPONENTS
   lldb-server
   repl_swift
   CACHE STRING "")
+
+# Begin Swift Mods
+set(LLDB_NO_INSTALL_DEFAULT_RPATH ON CACHE BOOL "")
```

Comment:
This is already [set upstream for macOS years back](https://github.com/llvm/llvm-project/blob/c565f09f4b0d908f51aaf4a841285f39ef93bc8c/lldb/cmake/caches/Apple-lldb-macOS.cmake#L4) so [bring it over for linux in the Swift build too](https://github.com/apple/swift/blob/1258d451523514d9ca140a9b7cb5a74463b7ca5e/utils/build-script-impl#L2126), [disabling adding](https://github.com/apple/llvm-project/blob/896de4fddf21190d40936b9f1ec24a71a9c370ed/lldb/cmake/modules/AddLLDB.cmake#L86) `$ORIGIN/../lib:/home/build-user/build/buildbot_linux/llvm-linux-x86_64/./lib` to all the lldb binaries shown above.

Reply:
Can this go upstream in `Apple-lldb-Linux.cmake`?

Reply:
Is there a need for it there? The reason it's needed in Swift's LLVM fork is because lldb is built separately from LLVM, which then triggers adding [the separate LLVM build directory to the rpath](https://github.com/apple/llvm-project/blob/232c77eede040008a4018c06f43a90655732faa3/llvm/cmake/modules/AddLLVM.cmake#L2306).

I'm not sure what these CMake config files are used for upstream, which normally builds everything together so doesn't have the rpath issue, so I can't say if this line would be useful there.